src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs

```diff
@@ -145,6 +162,74 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate? cert
             return AcquireCredentialsHandle(direction, secureCredential);
         }
 
+        public static unsafe SafeFreeCredentials AcquireCredentialsHandleNew(X509Certificate? certificate, SslProtocols protocols, EncryptionPolicy policy, bool isServer)
+        {
+            int protocolFlags = GetProtocolFlagsFromSslProtocols(protocols, isServer);
+            Interop.SspiCli.SCH_CREDENTIALS.Flags flags;
+            Interop.SspiCli.CredentialUse direction;
+
+            if (isServer)
+            {
+                direction = Interop.SspiCli.CredentialUse.SECPKG_CRED_INBOUND;
+                flags = Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_SEND_AUX_RECORD;
+            }
+            else
+            {
+                direction = Interop.SspiCli.CredentialUse.SECPKG_CRED_OUTBOUND;
+                flags =
+                    Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_CRED_MANUAL_CRED_VALIDATION |
+                    Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_CRED_NO_DEFAULT_CREDS |
+                    Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_SEND_AUX_RECORD;
+            }
+
+            if (policy == EncryptionPolicy.RequireEncryption)
+            {
+                // Always opt-in SCH_USE_STRONG_CRYPTO for TLS.
+                if (!isServer && ((protocolFlags & Interop.SChannel.SP_PROT_SSL3) == 0))
+                {
+                    flags |= Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_USE_STRONG_CRYPTO;
+                }
+            }
+            else if (policy == EncryptionPolicy.AllowNoEncryption)
+            {
+                // Allow null encryption cipher in addition to other ciphers.
+                flags |= Interop.SspiCli.SCH_CREDENTIALS.Flags.SCH_ALLOW_NULL_ENCRYPTION;
+            }
+            else
+            {
+                throw new ArgumentException(SR.Format(SR.net_invalid_enum, "EncryptionPolicy"), nameof(policy));
+            }
+
+            Interop.SspiCli.SCH_CREDENTIALS credential = default;
+            credential.dwVersion = Interop.SspiCli.SCH_CREDENTIALS.CurrentVersion;
+            credential.dwFlags = flags;
+
+            if (certificate != null)
+            {
+                credential.cCreds = 1;
+                Span<IntPtr> certificates = stackalloc IntPtr[1];
+                certificates[0] = certificate.Handle;
+                fixed (void* ptr = &certificates[0])
+                {
+                    credential.paCred = ptr;
+                }
+            }
```

Comment:
```suggestion
            IntPtr certificateHandle = IntPtr.Zero;
            if (certificate != null)
            {
                credential.cCreds = 1;
                certificateHandle = certificate.Handle;
                credential.paCred = &certificateHandle;
            }
```
?

Reply:
time will come when we need more than just one certificate to pass down custom chain. But since that is not needed for TLS13 I moved forward with proposed change. That should also make back-porting easier.